contracts/src/v0.8/functions/tests/1_0_0/FunctionsRouter.t.sol

```diff
@@ -1,111 +1,354 @@
+contract FunctionsRouter__OnlySenderThatAcceptedToS {
+
+}
+
+/// @notice #pause
+contract FunctionsRouter_Pause is FunctionsRouterSetup {
+  function setUp() public virtual override {
+    FunctionsRouterSetup.setUp();
   }
 
-  function getCoordinatorConfig() public pure returns (FunctionsBilling.Config memory) {
-    return
-      FunctionsBilling.Config({
-        maxCallbackGasLimit: 5,
-        feedStalenessSeconds: 5,
-        gasOverheadAfterCallback: 5,
-        gasOverheadBeforeCallback: 5,
-        requestTimeoutSeconds: 1,
-        donFee: 5,
-        maxSupportedRequestDataVersion: 5,
-        fulfillmentGasPriceOverEstimationBP: 5,
-        fallbackNativePerUnitLink: 2874
-      });
+  event Paused(address account);
+
+  function test_Pause_RevertIfNotOwner() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert("Only callable by owner");
+    s_functionsRouter.pause();
   }
 
-  function getTermsOfServiceConfig() public pure returns (TermsOfServiceAllowList.Config memory) {
-    return TermsOfServiceAllowList.Config({enabled: false, signerPublicKey: address(132)});
+  function test_Pause_Success() public {
+    vm.expectEmit(false, false, false, true);
+    emit Paused(OWNER_ADDRESS);
+
+    s_functionsRouter.pause();
+
+    bool isPaused = s_functionsRouter.paused();
+    assertEq(isPaused, true);
   }
 }
 
-contract FunctionsSetRoutes is FunctionsRouterSetup {
+/// @notice #unpause
+contract FunctionsRouter_Unpause is FunctionsRouterSetup {
   function setUp() public virtual override {
     FunctionsRouterSetup.setUp();
-    s_functionsCoordinator = new FunctionsCoordinator(
-      address(s_functionsRouter),
-      getCoordinatorConfig(),
-      address(s_linkEthFeed)
-    );
-
-    bytes32 allowListId = s_functionsRouter.getAllowListId();
-    bytes32[] memory proposedContractSetIds = new bytes32[](2);
-    proposedContractSetIds[0] = bytes32("1");
-    proposedContractSetIds[1] = allowListId;
-    address[] memory proposedContractSetAddresses = new address[](2);
-    proposedContractSetAddresses[0] = address(s_functionsCoordinator);
-    proposedContractSetAddresses[1] = address(s_termsOfServiceAllowList);
-
-    s_functionsRouter.proposeContractsUpdate(proposedContractSetIds, proposedContractSetAddresses);
-    s_functionsRouter.updateContracts();
+    s_functionsRouter.pause();
+  }
+
+  event Unpaused(address account);
+
+  function test_Unpause_RevertIfNotOwner() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert("Only callable by owner");
+    s_functionsRouter.unpause();
   }
+
+  function test_Unpause_Success() public {
+    vm.expectEmit(false, false, false, true);
+    emit Unpaused(OWNER_ADDRESS);
+
+    s_functionsRouter.unpause();
+
+    bool isPaused = s_functionsRouter.paused();
+    assertEq(isPaused, false);
+  }
+}
+
+// ================================================================
```

Comment:
Can subscription tests go to a separate file?

Reply:
Sure, yeah.